Resources/doc/first_steps.rst

```diff
@@ -0,0 +1,456 @@
+    }
+
+Updating an object involves just three steps:
+
+1. Fetching the object from Doctrine;
+2. Modifying the object;
+3. Calling ``flush()`` on the document manager.
+
+Notice that calling ``$dm->persist($product)`` isn't necessary. Recall that
+this method simply tells Doctrine to manage or "watch" the ``$product`` object.
+In this case, since you fetched the ``$product`` object from Doctrine, it's
+already managed.
+
+Deleting an Object
+~~~~~~~~~~~~~~~~~~
+
+Deleting an object is very similar, but requires a call to the ``remove()``
+method of the document manager:
+
+.. code-block:: php
+
+    $dm->remove($product);
+    $dm->flush();
+
+As you might expect, the ``remove()`` method notifies Doctrine that you'd
+like to remove the given document from the MongoDB. The actual delete operation
+however, isn't actually executed until the ``flush()`` method is called.
+
+Querying for Objects
+--------------------
+
+As you saw above, the built-in repository class allows you to query for one
+or many objects based on an number of different parameters. When this is
+enough, this is the easiest way to query for documents. Of course, you can
+also create more complex queries.
+
+Using the Query Builder
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Doctrine's ODM ships with a query "Builder" object, which allows you to construct
+a query for exactly which documents you want to return. If you use an IDE,
+you can also take advantage of auto-completion as you type the method names.
+From inside a controller:
+
+.. code-block:: php
+
+    $products = $this->get('doctrine_mongodb')
+        ->getManager()
+        ->createQueryBuilder('AcmeStoreBundle:Product')
+        ->field('name')->equals('foo')
+        ->sort('price', 'ASC')
+        ->limit(10)
+        ->getQuery()
+        ->execute()
+
+In this case, 10 products with a name of "foo", ordered from lowest price
+to highest price are returned.
+
+The ``QueryBuilder`` object contains every method necessary to build your
+query. For more information on Doctrine's Query Builder, consult Doctrine's
+`Query Builder`_ documentation. For a list of the available conditions you
+can place on the query, see the `Conditional Operators`_ documentation specifically.
+
+Custom Repository Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the previous section, you began constructing and using more complex queries
+from inside a controller. In order to isolate, test and reuse these queries,
+it's a good idea to create a custom repository class for your document and
+add methods with your query logic there.
+
+To do this, add the name of the repository class to your mapping definition.
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/Acme/StoreBundle/Document/Product.php
+        namespace Acme\StoreBundle\Document;
+
+        use Doctrine\ODM\MongoDB\Mapping\Annotations as MongoDB;
+
+        /**
+         * @MongoDB\Document(repositoryClass="Acme\StoreBundle\Repository\ProductRepository")

```

Comment:
We should use the `::class` constant instead of strings with class names.

Reply:
Same as previous comment, I'll be fixing these separately